core/src/main/java/com/linecorp/armeria/server/Routers.java

```diff
@@ -285,44 +315,51 @@
      */
     private static <V> Routed<V> findBest(RoutingContext routingCtx, @Nullable List<V> values,
                                           Function<V, Route> routeResolver) {
+        if (values == null) {
+            return Routed.empty();
+        }
+        return findBest(findAll(routingCtx, values, routeResolver, false));
```

Comment:
What's the expected performance hit we're gonna have? Doesn't this mean we have to get the complete list of routing results for all possible routes first?

Reply:
It's a naming problem. It actually converts all `ServiceConfig` instances into `Routed` instances, not getting all possible routes. Let me change the method name. :-)